### PR TITLE
chore: add better logging to setup action script

### PR DESCRIPTION
**Pull request by j-luong**
This PR should
- add timestamps to log output
- output expected and actual shasums when validation fails
- download binary and shasum from the same URL before retrying with backup URL
- fails download on any 400+ status code

### CODE REVIEW

File: setup/setup_snyk.sh [hunk 1]
```diff
@@ -49,48 +53,64 @@ esac
...
 #   $2: Output file name
 download_file() {
-    # Try to download from the main URL
-    if curl --compressed --retry 2 --output "$2" "$MAIN_URL/$1?utm_source="$GH_ACTIONS; then
-        echo "Downloaded from $MAIN_URL/$1"
-    # If main URL fails, try the backup URL
-    elif curl --compressed --retry 2 --output "$2" "$BACKUP_URL/$1?utm_source="$GH_ACTIONS; then
-        echo "Downloaded from $BACKUP_URL/$1"
-    # If both URLs fail, return an error
+    echo_with_timestamp "Downloading Snyk binary"
```
Comment (PeterSchafer) [inline setup/setup_snyk.sh:69]: nitpick: add download url already here, to know what is going to happen

```suggestion
    echo_with_timestamp "Downloading Snyk binary from $1"
```

**Review by PeterSchafer — APPROVED** [review]
LGTM!

**Review by sandor-trombitas — APPROVED** [review]

File: setup/setup_snyk.sh [hunk 2]
```diff
@@ -49,48 +53,64 @@ esac
...
+    if ! download_file "$BACKUP_URL/$VERSION" "snyk-${PREFIX}"; then
+        die "Failed to download Snyk files"
+    fi
 fi
 
-# Download SHA256 checksum file
-if ! download_file "$VERSION/snyk-${PREFIX}.sha256" "snyk-${PREFIX}.sha256"; then
-    die "Failed to download SHA256 file"
+# Verify the checksum
+echo_with_timestamp "Validating shasum"
```
Comment (PeterSchafer) [inline setup/setup_snyk.sh:96]: Suggestion: After some thought, I do have one suggestion. Could we move the shasum validation into download_file()? This way, we would attempt the fallback if the shasum fails. Also the operation download and shasum check seem reasonable to do together.

**Review by PeterSchafer — APPROVED** [review]
Nice!

**Pull request merged**